shared/nm-utils/nm-shared-utils.c

```diff
@@ -117,6 +117,37 @@ nm_utils_strbuf_append (char **buf, gsize *len, const char *format, ...)
 
 /*****************************************************************************/
 
+/**
+ * nm_utils_gbytes_equals:
+ * @bytes: (allow-none): a #GBytes array to compare. Note that
+ *   %NULL is treated like an #GBytes array of length zero.
+ * @arr: the data pointer with @arr_len bytes
+ * @arr_len: the length of the data pointer
+ *
+ * Returns: %TRUE if @bytes contains the same data as @arr. As a
+ *   special case, a %NULL @bytes is treated like an empty array.
+ */
+gboolean
+nm_utils_gbytes_equals (GBytes *bytes,
```

Comment:
I can't imagine this being used in a way that isn't entirely confusing and a major obstacle to readability.
I strongly suggest you just drop it.

Reply:
Since the function is used only in one place, I tend to agree that it would be better to open-code it.

The rest LGTM.